File: python/array_problem/pascal_triangle.py



def pascal_triangle(rawNumber):
    triangle = []  

    for i in range(rawNumber + 1):
        row = [1] * (i + 1) 
        for j in range(1, i):
            row[j] = triangle[i - 1][j - 1] + triangle[i - 1][j]
        triangle.append(row)  

    return triangle

File: js/src/ArrayProblem/PascalTriangle.js
export function PascalTriangle(rawNumber) {
    const triangle = [];

    for (let i = 0; i <= rawNumber; i++) {

    	triangle[i] = [];
        for (let j = 0; j <= i; j++) {

            if (j === 0 || j === i) {

                triangle[i][j] = 1; 

            } else {

                triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j];
            }
        }
    }

    return triangle;
}
